### PR TITLE
✨ webhook notifications

### LINKED ISSUE #295: Generic webhook notification
**Is your feature request related to a problem? Please describe.**
As the administrator of an application in the local travel space, I want ``camply`` to send a webhook (``HTTP/POST``) to a configuration-specified URL with configuration-specified headers and a JSON representation of the campsite so my application can programmatically handle the notification.

**Describe the solution you'd like**
When ``WEBHOOK_URL``, ``WEBHOOK_HEADERS``, and ``--notifications webhook`` are specified, the notification should be POST'ed to the ``WEBHOOK_URL`` with the ``WEBHOOK_HEADERS``. The request body should be JSON with machine-readable fields (example, dates should be epoch timestamps or other common time format).

Something like:

```json
{
  "campsite": {
    "campsite_id": "1234",
    "booking_date": 1694499820,
    "booking_url": "http://blahblah",
    "...": "etc"
  }
}
```

and

```
WEBHOOK_URL="http://foobar" \
  WEBHOOK_HEADERS=$(echo '{"Content-Type": "application/json", "x-api-key": "super-secret"}' | base64) \
  camply campsites --rec-area 12345
```

**Describe alternatives you've considered**

The open [Discord PR](https://github.com/juftin/camply/pull/280) is close, as is Slack, but the fields are human friendly or formatted, not generically machine-readable.

**Additional context**
If this sounds reasonable to you, I am willing to contribute a PR.

### PR DESCRIPTION
**Pull request by juftin**
# Description

Send notifications to a webhook endpoint

Resolves #295

### CODE REVIEW

File: camply/notifications/webhook.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    def send_campsites(self, campsites: List[AvailableCampsite], **kwargs):
+        """
+        Send a message with a campsite object
+
+        Parameters
+        ----------
+        campsites: List[AvailableCampsite]
+        """
+        self.send_message(message=campsites)
```
Comment (zdwolfe) [inline camply/notifications/webhook.py:69]: I recommend adding an envelope around ``campsites``, which allows for sibling data to be added in the future without making a backward incompatible change to the model.

So this:

```python
self.send_message(message=campsites)
```

could be

```python
self.send_message(message={"campsites": campsites})
```

which lets you hang non-campsite metadata or others if needed

```python

**Review by zdwolfe — CHANGES_REQUESTED** [review]
Very nice, you work quickly! I made one suggestion to avoid backward incompatible model changes in the future.

File: camply/notifications/webhook.py (hunk 1 above)
Comment (juftin) [reply camply/notifications/webhook.py]: Yeah, that's a good idea. I addressed this and added a `timestamp` field as well.

**Review by zdwolfe — APPROVED** [review]
LGTM!

**Pull request merged**